pkg/stanza/operator/input/windows/input.go

```diff
@@ -55,15 +101,25 @@ func (i *Input) Start(persister operator.Persister) error {
 		}
 	}
 
-	i.subscription = NewSubscription()
-	if err := i.subscription.Open(i.channel, i.startAt, i.bookmark); err != nil {
-		return fmt.Errorf("failed to open subscription: %w", err)
-	}
-
 	i.publisherCache = newPublisherCache()
 
+	var subscription Subscription
+	if i.isRemote() {
+		subscription = NewRemoteSubscription(i.remote.Server)
+		if err := subscription.Open(i.channel, i.startAt, i.bookmark, uintptr(i.remoteSessionHandle)); err != nil {
```

Comment:
How `subscription.Open` behaves if the remote server isn't reachable? Does it error? We need to becareful here because of the impact of failures on Start as mentioned by @djaglowski

Reply:
I updated the code to to test out bad credentials and a remote server that is not reachable. It will show the error every time depending on what the poll_interval is set to. 
Bad credentials error shows: `Access is denied.`
```log
2024-06-24T16:31:26.784Z        error   windows/input.go:110    Failed to open subscription for remote server   {"kind": "receiver", "name": "windowseventlog", "data_type": "logs", "operator_id": "windows_eventlog_input", "operator_type": "windows_eventlog_input", "server": "10.123.1.23", "error": "failed to subscribe to application channel: Access is denied."}
```

remote server that is not reachable shows: `The RPC server is unavailable.`
```log
2024-06-24T16:30:21.731Z        error   windows/input.go:110    Failed to open subscription for remote server   {"kind": "receiver", "name": "windowseventlog", "data_type": "logs", "operator_id": "windows_eventlog_input", "operator_type": "windows_eventlog_input", "server": "10.123.1.23", "error": "failed to subscribe to application channel: The RPC server is unavailable."}
```

I gave an attempt to mock the subscription, but I think it will take more time because bookmarks and events are also included. I'll give it another shot.